contracts/AaveMiddleware.sol

```diff
@@ -0,0 +1,217 @@
+      ).getLendingPool(); //Mainnet
+  }
+
+  function depositToken(uint256 _amount, address _tokenAddr) external {
+    IERC20 erc20Token = IERC20(_tokenAddr);
+    address _LendingPoolAddress = getLendingPoolAddress();
+    ILendingPool LendingPool = ILendingPool(_LendingPoolAddress);
+
+    erc20Token.safeTransferFrom(msg.sender, address(this), _amount); //Assuming the User has already Approved this amount using approve() function
+
+    erc20Token.safeApprove(_LendingPoolAddress, _amount);
+    LendingPool.deposit(_tokenAddr, _amount, address(this), 0); //The referral program is currently inactive and you can pass 0 as thereferralCode.
+  }
+
+  function withdrawToken(uint256 _amount, address _tokenAddr) external {
+    ILendingPool LendingPool = ILendingPool(address(getLendingPoolAddress()));
+    LendingPool.withdraw(_tokenAddr, _amount, msg.sender); //Amount Withdrawn to the user
+  }
+
+  function borrowToken(uint256 _amount, address _tokenAddr) external {
+    IERC20 erc20Token = IERC20(_tokenAddr);
+    address _LendingPoolAddress = getLendingPoolAddress();
+    ILendingPool LendingPool = ILendingPool(_LendingPoolAddress);
+    LendingPool.borrow(_tokenAddr, _amount, 2, 0, address(this)); //Amount borrowed. Will be received by the contract
+    erc20Token.safeTransfer(msg.sender, _amount); //Transferring the borrowed funds to the user
+  }
+
+  function repayToken(uint256 _amount, address _tokenAddr) external {
+    IERC20 erc20Token = IERC20(_tokenAddr);
+    address _LendingPoolAddress = getLendingPoolAddress();
+    ILendingPool LendingPool = ILendingPool(_LendingPoolAddress);
+    erc20Token.safeTransferFrom(msg.sender, address(this), _amount); //Assuming the User has already Approved this amount using approve() function
+    erc20Token.safeApprove(_LendingPoolAddress, _amount);
+    LendingPool.repay(_tokenAddr, _amount, 2, address(this)); //Amount borrowed
+  }
+
+  function depositEth() external payable {
+    address wethGatewayAddress = 0xcc9a0B7c43DC2a5F023Bb9b738E45B0Ef6B06E04; //Mainnet
+    address _LendingPoolAddress = getLendingPoolAddress();
+    IWETHGateway wethGateway = IWETHGateway(wethGatewayAddress);
+    wethGateway.depositETH{ value: msg.value }(
+      _LendingPoolAddress,
+      address(this),
+      0
+    );
+  }
+
+  function withdrawEth(uint256 _amount) external {
+    address wethGatewayAddress = 0xcc9a0B7c43DC2a5F023Bb9b738E45B0Ef6B06E04; //Mainnet
+    address _LendingPoolAddress = getLendingPoolAddress();
+    IWETHGateway wethGateway = IWETHGateway(wethGatewayAddress);
+    address aWethAddress = 0x030bA81f1c18d280636F32af80b9AAd02Cf0854e; //AWeth contract on Mainnet
+    IERC20 erc20Token = IERC20(aWethAddress);
+    erc20Token.safeApprove(wethGatewayAddress, _amount);
+    wethGateway.withdrawETH(_LendingPoolAddress, _amount, msg.sender);
+  }
+
+  function borrowEth(uint256 _amount) external payable {
+    address wethAddress = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2; //Weth on Ethereum
```

Comment:
Let's put this address as an immutable state variable.

Reply:
Changed